exporter/opentelemetry-exporter-datadog/src/opentelemetry/exporter/datadog/propagator.py

```diff
@@ -97,7 +97,7 @@ def inject(
             self.SAMPLING_PRIORITY_KEY,
             str(constants.AUTO_KEEP if sampled else constants.AUTO_REJECT),
         )
-        if constants.DD_ORIGIN in span.context.trace_state:
+        if span.context.trace_state.get(constants.DD_ORIGIN) is not None:
```

Comment:
I believe this will not happen from the latest version. Earlier `TraceState` was simple dict and there were no validation upon add/update key-value pair. We have fixed that in latest release and key-value pair much adhere to W3C tracestate format rules otherwise they will be discarded.

Reply:
Ah very nice.  I'll have to see if I can recreate the scenario I described in #262 with the current versions.  Though, I am a little concerned that by disallowing that to be `None` we'll have moved the error from this code I patched, to whatever code was actually causing it in the first place.

Reply:
Okay, rebased on the current main branch, and after some more testing, this PR still fixes the issue described in #262, so whatever change you're describing did not fix things.

Reply:
That's strange that you are still running into that error. From [here](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-api/src/opentelemetry/util/tracestate.py#L66) every value in the tracestate must be a str type.

Reply:
Yeah, and I still haven't figured out where it's coming from.

Reply:
Did you update the `opentelemetry-api` and propagator versions you are using? Can you provide a reproducible example?

Reply:
Yep, this is with code from the main branch of both main and contrib repositories.

I'll see if I can strip down something - so far I've only seen when making gRPC client calls to a gRPC service running the botocore instrumentation and datadog exporter, so it's a little complex to reproduce, but I might be able to simplify more.  Give me a day or two.